src/systems/systems/src/hydraulic/nose_steering.rs

```diff
@@ -0,0 +1,678 @@
+
+    const MAX_SPEED_FILTER_TIMECONST: Duration = Duration::from_millis(100);
+    const CURRENT_SPEED_FILTER_TIMECONST: Duration = Duration::from_millis(150);
+
+    pub fn new(
+        context: &mut InitContext,
+        max_half_angle: Angle,
+        nominal_speed: AngularVelocity,
+        actuator_diamter: Length,
+        angular_to_linear_ratio: Ratio,
+    ) -> Self {
+        Self {
+            position_id: context.get_identifier("NOSE_WHEEL_POSITION".to_owned()),
+
+            current_speed: LowPassFilter::<AngularVelocity>::new(
+                Self::CURRENT_SPEED_FILTER_TIMECONST,
+            ),
+            current_position: Angle::new::<radian>(0.),
+
+            max_half_angle,
+
+            max_speed: LowPassFilter::<AngularVelocity>::new(Self::MAX_SPEED_FILTER_TIMECONST),
+            nominal_speed,
+            angular_to_linear_ratio,
+
+            total_volume_to_actuator: Volume::new::<gallon>(0.),
+            total_volume_to_reservoir: Volume::new::<gallon>(0.),
+
+            actuator_area: std::f64::consts::PI * (actuator_diamter / 2.) * (actuator_diamter / 2.),
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        current_pressure: Pressure,
+        steering_controller: &impl SteeringController,
+        pushback_tug: &impl Pushback,
+    ) {
+        if !pushback_tug.is_nose_wheel_steering_pin_inserted() {
+            self.update_max_speed(context, current_pressure);
+
+            let limited_requested_angle = steering_controller
+                .requested_position()
+                .min(self.max_half_angle)
+                .max(-self.max_half_angle);
+
+            self.update_current_speed(context, limited_requested_angle);
+
+            self.update_final_speed_position(context, limited_requested_angle);
+        } else {
+            self.update_speed_position_during_pushback(pushback_tug);
+        }
+
+        self.update_flow(context, pushback_tug);
+    }
+
+    fn update_final_speed_position(&mut self, context: &UpdateContext, requested_angle: Angle) {
+        self.current_position += Angle::new::<radian>(
+            self.current_speed.output().get::<radian_per_second>() * context.delta_as_secs_f64(),
+        );
+
+        let position_error_abs = (requested_angle - self.position_feedback()).abs();
+
+        // If we crossed desired position between frames we assume we stopped at correct position
+        // Checking a position error because if requested angle changed since last frame by a huge
+        // amount, we want to track new position not to directly set new position
+        if position_error_abs.get::<degree>() < 5.
+            && (self.current_speed.output().get::<radian_per_second>() > 0.
+                && requested_angle < self.position_feedback()
+                || self.current_speed.output().get::<radian_per_second>() < 0.
+                    && requested_angle > self.position_feedback())
+        {
```

Comment:
**suggestion**: For extra clarity you might want to consider adding parentheses. I always find that having to think about expression precedence while reading expressions is a mental burden that doesn't have to be there.

Reply:
I think the same and only do this because I thought I would be clippified for not needed parenthesis :D
Will try to ass them :p

Reply:
Well could add those but auto formatting doesn't help to make it more readable sadly